### ISSUE TITLE
RFE: Directory structure for plugin antora/asciidoc files 

### ISSUE THREAD

**Issue by Rasbats** [Rasbats/shipdriver_pi#159]
At present the antora.yml used for building the plugin user guide can be found in the docs folder. Also in this folder is modules/ROOT which holds the asciidoc files.

This structure simplifies placing java code in the docs folder to enable building a pdf for the guide.

**Comment by Rasbats** [comment]
An example of using java to create a pdf from asciidoc is here:

[https://github.com/Rasbats/otidalplan_pi/tree/master/docs](https://github.com/Rasbats/otidalplan_pi/tree/master/docs)

**Comment by leamas** [comment]
My thoughts so far:

  - No-one here probably has the experience to say anything for sure.
  - The directory structure in the plugin, which also is a Antora component, shouldn't really affect for example java generation. The Antora way is to build in a separate playbook directory rather than in the component.
  - There is no need no standardize at this point. It works either way, and it might be valuable to explore dfferent paths.
  - *If* you want to have the manual stuff in a singe directory, the name *docs* is a bit unfortunate. One of the most common ways to deploy the results is with Github Pages (GP). GP has hardcoded directory name *docs* which is used for the Antora **output** i. e., the formatted docs. It will work anyway, since we should run Antore i a separate playbook dir. But using the name *docs* for the Antora input, when it's sort of an established name for the output will confuse some users.
  - There seems to be several ways to convert to pdf, none of them really mature. The upcoming way using Antora tools seems to be https://gitlab.com/djencks/antora-extension-examples/-/tree/master/antora-pdf/simple-pdfs. That said, many users seems to go ways similar to what you describe, based on direct adoc -> pdf conversion.
  - The need for pdf is perceived as a upper-management  issue in many cases, I dare say the needs of an  older generation still focused on paper documentation. Practically, in a yacht, a downloaded HTML manual might serve user needs as well or better than a PDF version.

**Comment by leamas** [comment]
Hm... looking into what you have done with managed-plugins. Of course, converting this to asciidoc is great, That said, you have ended up with a docs/ dir which is both sources (antora.yml, modules/, ...) and the generated site (public/). This is extremely messy, and not something I think is sound. Furthermore, it will not work with Github Pages.

The immediate fix is to not run antora in the component i. e., this repo but in a separate playbook dir. This way, there will be a fresh docs/  directory separate from the source directory in this repo.

Next step to clean things up could be to just move antora.yml and modules to the top directory. Or, rename the docs dir to something else, perhaps 'manual'.

**Comment by Rasbats** [comment]
> The upcoming way using Antora tools seems to be https://gitlab.com/djencks/antora-extension-examples/-/tree/master/antora-pdf/simple-pdfs

That is the source I used. When the plugin manual is in adoc format and living in modules/ROOT it can be referenced instead of getting it from the wiki. The convoluted structure you see at the moment can change. I was playing with python;),

Happy to rename my root/docs directory to 'manual'! 

>  looking into what you have done with managed-plugins

Did you mean 'odtidalplan'?

The build stuff in shipdriver/docs (including site.yml) was the result of playing with a local build as part of learning how to make the reference to docs in ocpn_manuals. It needs removing.

**Comment by leamas** [comment]
>The build stuff in shipdriver/docs (including site.yml) was the result of playing with a local build as part of learning how to make the reference to docs in ocpn_manuals. It needs removing.

This is interesting. I'm trying to collect all known modules in ocpn-manuals at this point. The idea is that one we have everything on one place we can start the work of re-organizing and reviewing what we have; there is an obvious need for that.

OTOH, I don't really understand what the plugin manuals does in the developer's manual. To me, these looks like typical end-user stuff.

The most important piece is the managed-plugins part. I have been able to include it as-is. Please let me know if you change the interface (module name, start_dir, etc.)

About shipdriver: should we try to include in ocpn-manuals, or drop it for now?

**Comment by leamas** [comment]
Off-topic: This is basically a de-centralized work, we work on different modules. This is actually great.

But: we need to agree on the central clearing-house which collects all modules and builds the overall manual. My proposal is that we use leamas/ocpn-manuals for this. 

Thougts?

**Comment by Rasbats** [comment]
> I don't really understand what the plugin manuals does in the developer's manual

Me neither! It was the result of your use of the ShipDriver manual in the build of ocpn_manuals.

> About shipdriver: should we try to include in ocpn-manuals, or drop it for now?

Please drop it. Intend moving forward with the ocpn_user_manual when time allows. Need to get the route planning pi working correctly.

> clearing-house which collects all modules and builds the overall manual. My proposal is that we use leamas/ocpn-manuals for this.

Agreed. 

> Or, rename the docs dir to something else, perhaps 'manual'.

If it doesn't screw things up I will change docs to 'manual' in the asciidocs branch of Rasbats/managed_plugins.

Ref:[https://github.com/OpenCPN/OpenCPN/issues/2071](https://github.com/OpenCPN/OpenCPN/issues/2071)

Do we continue with this or could Rick and I continue by raising issues in leamas/ocpn-manuals?

**Comment by leamas** [comment]
> Me neither! It was the result of your use of the ShipDriver manual in the build of ocpn_manuals.

Which was caused by the plugins manuals being included in the dokuwiki developer manual. Up to now, I have been porting it more or less mechanically, but you're right. It's time to change that.

> Please drop it. Intend moving forward with the ocpn_user_manual when time allows. Need to get the route planning pi working correctly.

OK

> If it doesn't screw things up I will change docs to 'manual' in the asciidocs branch of Rasbats/managed_plugins.

If/when you do, please make a corresponding PR to ocpn-manuals so we can keep it in sync.

> Do we continue with this or could Rick and I continue by raising issues in leamas/ocpn-manuals?

That one has become a general discussion forum on this. Somehow we need this, that bug is already swamped with the discussion... perhaps best to use it instead of messing up some other issue for now, until we found out a new way to organize the discussion

**Comment by rgleason** [comment]
Alec wrote

`Which was caused by the plugins manuals being included in the dokuwiki developer manual. Up to now, I have been porting it more or less mechanically, but you're right. It's time to change that.`

Those plugins are Beta Plugins, about 10-13 of them.  

I hope you do not intend to include all 41-42 normal plugins in the User Documentation. It is a sizeable chunk of data and media. It should be distributed individually for offline use to reduce the overall size of the User Manual when distributed for offline use.  When the User Manual is on-line, there could be a page with external links to each of the plugins output documentation (html or pdf) where ever that is required to reside or when the plugin is updated, the right format documentation could land in a dedicated spot for the online manual.

I also think the plugin "manual"  directory should reside in the github.com/[user]/[plugin_pi]/manual directory so that the antora commands can include the right format output with each tarball.  There may be a need to have output also go to some dedicated location to update the on line User Manual and some other location for offline to get into the tarball.

Just some thoughts here.   I think html is the right format for online documentation.   It could also be used for offline. I know that we used to use html for the User Manual some years ago.
